src/parser/ast.rs

```diff
@@ -205,13 +206,19 @@ pub enum Type<'a> {
     Boolean(Location<'a>),
     Unit(Location<'a>),
     Reference(Location<'a>),
-    Function(Vec<Type<'a>>, Box<Type<'a>>, /*varargs:*/ bool, /*closure*/ bool, Location<'a>),
+    Function(Vec<Type<'a>>, Box<Type<'a>>, Vec<Effect<'a>>, /*varargs:*/ bool, /*closure*/ bool, Location<'a>),
```

Comment:
Is non-ordered container preferable?

Reply:
Likely, but this is fine as a first pass. The order of effect annotations doesn't affect anything, but the order of effect handlers in the code can affect the output.